src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2694,7 +2696,8 @@ ReceiveResults(WorkerSession *session, bool storeRows)
 		if (result == NULL)
 		{
 			/* no more results */
-			return true;
+			fetchDone = true;
```

Comment:
maybe `no more results` -> `no more results, do not immediately return because we should free some memory before` (or probably a better comment)

because it is not obvious that why we're deferring the return here.

Reply:
done